app/services/foreman_bootdisk/iso_generator.rb

```diff
@@ -24,42 +25,60 @@ def self.generate_full_host(host, opts = {}, &block)
         hash[iso_filename] = host.url_for_boot(type)
       end
 
-      generate(opts.merge(isolinux: tmpl, files: files), &block)
+      generate(opts.merge(isolinux: isolinux_template, grub: grub_template, files: files), &block)
     end
 
-    def self.render_pxelinux_template(host)
-      pxelinux_template = host.provisioning_template(kind: :PXELinux)
+    def self.render_template(host, kind, scope_class)
+      template = host.provisioning_template(kind: kind)
 
-      raise(Foreman::Exception, N_('Unable to generate disk template, PXELinux template not found.')) unless pxelinux_template
+      raise Foreman::Exception.new(N_('Unable to generate disk template, %{kind} template not found.'), kind: kind) unless template
 
       template = ForemanBootdisk::Renderer.new.render_template(
-        template: pxelinux_template,
+        template: template,
         host: host,
-        scope_class: ForemanBootdisk::Scope::FullHostBootdisk
+        scope_class: scope_class
       )
 
       unless template
         err = host.errors.full_messages.to_sentence
-        raise(::Foreman::Exception.new(N_('Unable to generate disk PXELinux template: %s'), err))
+        raise Foreman::Exception.new(N_('Unable to generate disk %{kind} template: %{error}'), kind: kind, error: err)
       end
 
       template
     end
 
+    def self.system_or_exception(command, error)
+      unless system(command)
```

Comment:
In another PR we should refactor this to properly capture and log (for debugging purposes) the output of these commands.

```ruby
require 'open3'
stdout, stderr, status = Open3.capture3(command)
logger.debug "Whatever #{stdout}, ..."
```

Reply:
Agreed, perhaps for all commands issued. Maybe something in Foreman core even.